source/aws-bootstrap-kit/lib/account.ts

```diff
@@ -84,72 +90,114 @@ export class Account extends Construct {
   readonly accountName: string;
   readonly accountId: string;
   readonly accountStageName?: string;
+  readonly accountStageOrder?: number;
 
   constructor(scope: Construct, id: string, accountProps: IAccountProps) {
     super(scope, id);
 
     const accountProvider = AccountProvider.getOrCreate(this);
 
-    let account = new core.CustomResource(
-      this,
-      `Account-${accountProps.name}`,
-      {
-        serviceToken: accountProvider.provider.serviceToken,
-        resourceType: "Custom::AccountCreation",
-        properties: {
-          Email: accountProps.email,
-          AccountName: accountProps.name,
-          AccountType: accountProps.type,
-          StageName: accountProps.stageName,
-          StageOrder: accountProps.stageOrder?.toString(),
-          HostedServices: accountProps.hostedServices?accountProps.hostedServices.join(':'):undefined
+    let existingAccount = false;
+    let accountId = accountProps.id;
+    let hostedServices = accountProps.hostedServices ? accountProps.hostedServices.join(':') : undefined;
+
+    // do not create account if we reuse one
+    let account;
+    if (!accountId) {
+      account = new core.CustomResource(
+        this,
+        `Account-${accountProps.name}`,
+        {
+          serviceToken: accountProvider.provider.serviceToken,
+          resourceType: "Custom::AccountCreation",
+          properties: {
+            Email: accountProps.email,
+            AccountName: accountProps.name,
+          },
+          removalPolicy: accountProps.removalPolicy || RemovalPolicy.RETAIN
         },
-      }
-    );
+      );
+      accountId = account.getAtt("AccountId").toString();
+      accountProps.id = accountId;
+    } else {
+      existingAccount = true;
 
-    let accountId = account.getAtt("AccountId").toString();
+      // retrieve existing account information (actual name and email) to update accountProps
+      account = new cr.AwsCustomResource(this, "ExistingAccountCustomResource", {
+        onUpdate: {
+          service: "Organizations",
+          action: "describeAccount",
+          physicalResourceId: cr.PhysicalResourceId.fromResponse(
+            "Account.Id"
+          ),
+          region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+          parameters: {
+            AccountId: accountId
+          }
+        },
+        policy: cr.AwsCustomResourcePolicy.fromSdkCalls({
+          resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE,
+        })
+      });
+
+      accountProps.name = account.getResponseField("Account.Name");
+      accountProps.email = account.getResponseField("Account.Email");
+    }
+
+    // add tags
```

Comment:
To support Tags I have a different idea. The account needs to implement the ITaggable and should have a property tags of type Tagmanager. Then we need to fetch the current tags. Calculate the diff and call tagResource/untagResource

https://github.com/pepperize/cdk-organizations/blob/693ecef594b4d7c093f48b37b84e97ddbdd70fbc/src/tag-resource-provider/on-event-handler.lambda.ts#L11

Reply:
Good idea (https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.TagManager.html). But maybe not in this PR.

If this is implemented in cdk-org, and we decide to move to it, will be great.